core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -183,6 +185,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     val updateMetadataResponse =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         replicaManager.maybeUpdateMetadataCache(correlationId, updateMetadataRequest, metadataCache)
+        if (controller.isActive && !updateMetadataRequest.partitionStates.isEmpty) {
```

Comment:
I removed the controller.isActive check and replaced it with a hasDelayedTopicOperations check to maintain the benefit of not iterating over all of the topics if the purgatory is empty/unused.